### ISSUE TITLE
Dump debug information

### ISSUE THREAD

**Issue by 0xc0170** [project-generator/project_generator#310]
We provide debug project information printed (export/tool_specific/common). For bigger projects, the there's a lot of information, not much readable. Therefore I am considering adding a log file, in more readable form (yaml for instance). I'll test the outline. In case of problems, this log can pinpoint a problem.

**Comment by 0xc0170** [comment]
While fixing some tests, this is an output, much more readable (it's show on the console as well).

```
common:
  build_dir: build
  debugger: j-link
  export_dir: ''
  includes: &id001
  - includes: [test_workspace/header1.h]
  linker_file: test_workspace/linker.ld
  macros: &id002 [MACRO1, MACRO2]
  misc: &id003 {}
  name: project_3
  output_type: exe
  sources: &id004
  - sources: [test_workspace/main.cpp]
  target: mbed-lpc1768
  template: ''
  tools_supported: &id005 [iar_arm, uvision, coide, unknown]
merged:
  build_dir: build
  debugger: j-link
  export_dir: ''
  include_files:
    includes: [../../test_workspace/header1.h]
  include_paths: [../../test_workspace]
  includes: *id001
  linker_file: ../../test_workspace/linker.ld
  macros: *id002
  misc: *id003
  name: project_3
  output_dir: {path: generated_projects/iar_arm_project_3, rel_count: 2, rel_path: ../../}
  output_type: exe
  singular: true
  source_files_c: {}
  source_files_cpp:
    sources: [../../test_workspace/main.cpp]
  source_files_lib: {}
  source_files_obj: {}
  source_files_s: {}
  source_paths: [../../test_workspace]
  sources: *id004
  target: mbed-lpc1768
  template: []
  tools_supported: *id005
tool_specific: {}
```

Dumpted to name.progen.log file

**Comment by 0xc0170** [comment]
Merged to v0.8
